metadata/src/main/java/org/apache/kafka/controller/PartitionChangeBuilder.java

```diff
@@ -361,11 +410,16 @@ public Optional<ApiMessageAndVersion> build() {
 
         maybeUpdateRecordElr(record);
 
-        if (record.isr() == null && !targetIsr.isEmpty() && !targetIsr.equals(Replicas.toList(partition.isr))) {
+        // If ELR is enabled, the ISR is allowed to be empty.
+        if (record.isr() == null && (!targetIsr.isEmpty() || eligibleLeaderReplicasEnabled) && !targetIsr.equals(Replicas.toList(partition.isr))) {
             // Set the new ISR if it is different from the current ISR and unclean leader election didn't already set it.
+            if (targetIsr.isEmpty()) {
+                log.info("A partition will have an empty ISR. " + this);
```

Comment:
I think we might want DEBUG here. If we are emptying out the ISR, it's not necessarily a problem that requires user intervention. Since we now allow emptying the ISR when the final replica goes down, we would end up logging this for every partition if someone did a full cluster shutdown.

Reply:
Good point. log will be flushed with this message during a shutdown.